README.md

```diff
@@ -1,72 +1,166 @@
-# webboot
-Tools to let a u-root instance boot signed live distro images over the web
+# About
 
-## How to use webboot
+webboot offers tools to let a u-root instance boot signed live distro images
+over the web.
 
-### Build initramfs with added webboot commands.
-- Run the command `go run buildimage.go` in the home directory of webboot to build initramfs.
+## Idea
 
-### Boot into Qemu 
-Tip:Dont use the -nographic option on qemu when booting u-root in qemu.
- `qemu-system-x86_64 -kernel arch/x86/boot/bzImage  -append 'console=ttyS0 console=tty1' -initrd /tmp/initramfs.linux_amd64.cpio -enable-kvm -m 20G  -device virtio-rng-pci`
+The key idea is to have the respective distro's image copied to RAM and preserve
+it throughout a kexec. That is achieved by using a pmem device, which is created
+at `/dev/pmem[N]` when booting Linux with the `memmap` parameter.
 
-### Run the webboot command in Qemu
+The caveat is that both our webboot kernel as well as the kernel we kexec into
+need support for pmem.
 
- `webboot distribution`
+For reference, webboot developers should familiarize themselves with:
 
-- distribution a defines a supported operating system distribution.
-- example `webboot tinycore`perfoms a webboot of the TinyCore linux distribution.
+- [cpio tutorial](https://www.gnu.org/software/cpio/manual/html_node/Tutorial.html)
+- [initrd usage](https://www.kernel.org/doc/html/latest/admin-guide/initrd.html)
+- [kernel parameters](https://www.kernel.org/doc/html/latest/admin-guide/kernel-parameters.html)
+- the [memmap option](https://docs.pmem.io/persistent-memory/getting-started-guide/creating-development-environments/linux-environments/linux-memmap)
 
-### Supported Operating Systems
-- Linux TinyCore
+## Usage
+
+### Build initramfs with added webboot commands
+
+Run `go run buildimage.go` in the source directory of webboot to build the
+initramfs.
+
```

Comment:
you need some other commands to CORRECTLY build initramfs. Heres what you need: 

go run buildimage.go -u '-build=bb -files $HOME/linux/arch/x86/boot/bzImage:bzImage' 

:bzImage puts bzImage in the initramfs, and before that just needs to be a path to the kernel